lib/commands/exam.js

```diff
@@ -206,50 +217,75 @@ module.exports = TestCommand.extend({
    */
   _generateCustomConfigs(commandOptions) {
     const config = this._super._generateCustomConfigs.apply(this, arguments);
+    const events = config.custom_browser_socket_events || {};
+
+    this._addBrowserSocketEvents(config, events);
```

Comment:
instead of modifying an input, can we have a pure function that returns an object with the events?
```
_generateCustomConfigs(commandOptions) {
    const config = this._super._generateCustomConfigs.apply(this, arguments);
    let additionalEvents =  this._getBrowserSocketEvents(config);

    if (commandOptions.loadBalance || commandOptions.replayExecution) {
      const loadBalancingEvents = this._getLoadBalancingBrowserSocketEvents(this.commands, this.testemEvents);
      additionalEvents = { ...additionalEvents, ...loadBalancingEvents };
    }

    config.custom_browser_socket_events = { ...config.custom_browser_socket_events, ...additionalEvents };
```